packages/component-library/components/Box/Box.tsx

```diff
@@ -0,0 +1,42 @@
+import classnames from "classnames"
+import * as React from "react"
+import { responsiveMarginClasses } from "../Spacing/margin"
+import { responsivePaddingClasses } from "../Spacing/padding"
+import { Spacing } from "../types"
+
+export interface BoxProps {
+  children: React.ReactNode
+  classNameAndIHaveSpokenToDST?: string
+}
+
+export const Box = ({
+  children,
+  classNameAndIHaveSpokenToDST,
+  p,
```

Comment:
I've borrowed the naming conventions from `styled-system`: https://styled-system.com/api#space

A couple of thoughts:
* Should we allow long names too? (m and margin, pl and paddingLeft, etc?)
* Should our grid accept negative numbers?

Reply:
The plus side of short names only is that you won't get confused with actual `style` attribute keys, which use the long names, but expect css strings.

Reply:
I don't think a `Box` should have negative values. It's usually a hack to add gutters to grids or add "aesthetic overlapping", and both of those cases are either rare or very specific.

Reply:
@kemblesong Yeah, that's a really good call out. As much as I like the idea of allowing both naming conventions I think it may cause confusion.

Reply:
I might be old fashioned but I prefer the long names 😅 I find less thinking is involved when reading code, especially if I'm new to the API.

But this is total bikeshedding, feel free to disregard